### PR TITLE
RTSP Fingerprints Added

### PR DESCRIPTION
**Pull request by cbarnard-r7**
## Description
This PR adds signatures for RTSP servers.

## Motivation and Context
These signatures were created after building new studies in Sonar, giving us a large volume of RTSP banners to analyze.

## How Has This Been Tested?
These signatures were run through the results of an RTSP Sonar study. Passed recog_verify

## Types of changes

- Non-breaking updates to fingerprints

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by cbarnard-r7** [comment]
I created a new database for the RTSP sigs. It is notable that the RTSP and HTTP protocols are nearly identical, and an HTTP request to an RTSP server will return the necessary information for recog to work on these fingerprints.

It's possible that these could (should?) be added to one of the HTTP databases, so incidental queries to RTSP servers through HTTP requests would still get successfully fingerprinted, but for the time being, it felt reasonable to keep them separate.

**Comment by hdm** [comment]
Hi folks! I have been working on RTSP coverage as well and had merged these into http_servers.xml.
I had not submitted these yet, as I waiting on #249 to merge. 

You can find the full comparison from my development branch at: https://github.com/rapid7/recog/compare/master...hdm:development

**Comment by hdm** [comment]
This commit in particular adds both SIP and RTSP header matching to http_servers.xml: https://github.com/rapid7/recog/commit/f62549dd6ef3ccc1f9e01734cb9a722623f88b9a

**Comment by hdm** [comment]
I rolled up our recent SIP/RTSP into this PR: https://github.com/rapid7/recog/pull/254

Happy to help merge these if you like!

**Comment by tsellers-r7** [comment]
@hdm  and @cbarnard-r7  - Let's move the fingerprints for RTSP into `http_servers.xml`. Just remember to set `service.protocol` on anything RTSP specific.

**Comment by hdm** [comment]
Got it! Does that apply to SIP as well?

**Comment by tsellers-r7** [comment]
No, we will leave `sip_banners.xml` as it is for now.

**Comment by hdm** [comment]
It looks like RTSP may need to be its own after all, otherwise the fingerprints collide with HTTP:
```
rspec './spec/lib/fingerprint_self_test_spec.rb[1:10:906:7]' # Recog::DB#http_servers.xml (?-mix:^WMServer\/([\d\.]+)$) Example 'WMServer/9.1.1.3862' matches this regex first

rspec './spec/lib/fingerprint_self_test_spec.rb[1:10:906:10]' # Recog::DB#http_servers.xml (?-mix:^WMServer\/([\d\.]+)$) Example 'WMServer/9.5.6001.22609' matches this regex first
```

**Comment by hdm** [comment]
@cbarnard-r7 could you add this fingerprint your PR?
```
  <fingerprint pattern="^Cisco MediaSense Media Server$">
    <description>Cisco MediaSense Media Server (RTSP)</description>
    <example>Cisco MediaSense Media Server</example>
    <param pos="0" name="service.vendor" value="Cisco"/>
    <param pos="0" name="service.product" value="MediaSense Media Server"/>
    <param pos="0" name="service.protocol" value="rtsp"/>
    <param pos="0" name="os.vendor" value="Cisco"/>
    <param pos="0" name="os.family" value="Linux"/>
    <param pos="0" name="os.product" value="VOS"/>
    <param pos="0" name="hw.vendor" value="Cisco"/>
    <param pos="0" name="hw.device" value="SIP Gateway"/>
    <param pos="0" name="hw.product" value="MediaSense Server"/>
  </fingerprint>
```

**Comment by cbarnard-r7** [comment]
Yeah, I can put these in my PR.

I understand RTSP and HTTP are super similar, but I personally feel like keeping them bumped out is the safest route, here. While there may only be one or two collisions with these fingerprints, conflating protocols like this feels like something that will cause issues down the line.

**Comment by hdm** [comment]
Thanks! It looks like that was the only RTSP from my side (the rest were SIP, now broke into a separate PR). This changeset looks good here.

File: xml/rtsp_servers.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<fingerprints matches="rtsp_header.server" protocol="rtsp" database_type="service">
```
Comment (tsellers-r7) [inline xml/rtsp_servers.xml]: Please add `preference="0.85"` which will put it under `http_banners.xml` in priority.

Comment (cbarnard-r7) [reply xml/rtsp_servers.xml]: Done.

**Comment by tsellers-r7** [comment]
I'm going to go ahead and land this PR.  I'll commit the requested changes above afterwards.

**Pull request merged**